src/shims/unix/linux/foreign_items.rs

```diff
@@ -79,13 +79,16 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             "pthread_setname_np" => {
                 let [thread, name] =
                     this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
-                let res = this.pthread_setname_np(
+                let res = match this.pthread_setname_np(
                     this.read_scalar(thread)?,
                     this.read_scalar(name)?,
                     TASK_COMM_LEN,
                     /* truncate */ false,
-                )?;
-                let res = if res { Scalar::from_u32(0) } else { this.eval_libc("ERANGE") };
+                )? {
+                    ThreadNameResult::Ok => Scalar::from_u32(0),
+                    ThreadNameResult::NameTooLong => this.eval_libc("ERANGE"),
+                    ThreadNameResult::ThreadNotFound => this.eval_libc("ENOENT"),
```

Comment:
I don't see ENOENT mentioned in the man page. Where did you get that error code from?

Reply:
That's again the same problem I told you about in the corresponding issue. Not every documentation is perfect (:

Reply:
Well you got the code from somewhere. So please add comments explaining where you got it from.

Reply:
Ah, wrong place, wasn't able to see the file you left the comment for, so thought that you are about FREEBASE or something else.

For Linux it's quite clear, the man pages state that all error codes unrelated to the buffer length are from `open` system call. For others there are links to the source already.

Reply:
> For Linux it's quite clear, the man pages state that all error codes unrelated to the buffer length are from open system call. 

Ah good point, I had missed that.